src/spikeinterface/sorters/internal/spyking_circus2.py

```diff
@@ -357,27 +356,44 @@ def _run_from_folder(cls, sorter_output_folder, params, verbose):
         return sorting
 
 
-def final_cleaning_circus(recording, sorting, templates, **merging_kwargs):
-
-    from spikeinterface.core.sorting_tools import apply_merges_to_sorting
-    from spikeinterface.curation.curation_tools import resolve_merging_graph
-
+def create_sorting_analyzer_with_templates(sorting, recording, templates, remove_empty=True):
     sparsity = templates.sparsity
     templates_array = templates.get_dense_templates().copy()
 
-    sa = create_sorting_analyzer(sorting, recording, format="memory", sparsity=sparsity)
+    if remove_empty:
+        from spikeinterface.core.sparsity import ChannelSparsity
+
+        non_empty_unit_ids = sorting.get_non_empty_unit_ids()
+        non_empty_sorting = sorting.remove_empty_units()
+        non_empty_unit_indices = sorting.ids_to_indices(non_empty_unit_ids)
+        templates_array = templates_array[non_empty_unit_indices]
+        sparsity_mask = sparsity.mask[non_empty_unit_indices, :]
+        sparsity = ChannelSparsity(sparsity_mask, non_empty_unit_ids, sparsity.channel_ids)
+    else:
+        non_empty_sorting = sorting
 
+    sa = create_sorting_analyzer(non_empty_sorting, recording, format="memory", sparsity=sparsity)
```

Comment:
you should put job_kwargs

Reply:
With the sparsity given it doesn't matter